src/MetricsUploader/include/MetricsUploader/MetricsUploader.h

```diff
@@ -36,39 +36,20 @@ const std::string kMetricsUploaderClientDllName = "metrics_uploader_client";
 //
 class MetricsUploader {
  public:
+  MetricsUploader() = default;
+  virtual ~MetricsUploader() = default;
+
   // Create a MetricsUploader instance, load metrics uploader client library if available and starts
   // metrics uploader client when called first time. Return the instance if there are no errors and
   // ErrorMessage otherwise.
-  [[nodiscard]] static ErrorMessageOr<MetricsUploader> CreateMetricsUploader(
+  [[nodiscard]] static ErrorMessageOr<std::unique_ptr<MetricsUploader>> CreateMetricsUploader(
```

Comment:
Why it's better to have a `unique_ptr` here instead of value?

Reply:
Its not better, its necessary because of the pure virtual situation (see other comment). Indeed it would be nicer to just have the value and not a pointer.